src/compiler/checker.ts

```diff
@@ -24520,6 +24528,10 @@ namespace ts {
                 propType = indexInfo.type;
             }
             else {
+                if (prop?.isDeprecated) {

```

Comment:
Same, it's inside `if (!prop) ... else`

```suggestion
                if (prop.isDeprecated) {
```